units/commit_queue.go

```diff
@@ -66,6 +66,32 @@ func NewCommitQueueJob(env evergreen.Environment, queueID string, id string) amb
 	return job
 }
 
+func (j *commitQueueJob) TryUnstick(cq *commitqueue.CommitQueue) {
+	//unstuck the queue if the patch is done.
+	nextItem := cq.Next()
+	if nextItem != nil {
+		patch, err := patch.FindOne(patch.ById(patch.NewId(nextItem.Issue)).WithFields(patch.FinishTimeKey))
+		if err != nil {
+			j.AddError(errors.Wrapf(err, "error determining if patch is done for %s", j.QueueID))
+		}
+
+		//patchisdone
+		if !patch.FinishTime.IsZero() {
```

Comment:
Does returning here eliminate that concern? 
if err != nil {
			j.AddError(errors.Wrapf(err, "error determining if patch is done for %s", j.QueueID))
			return
		}

Reply:
I think it would still be an issue: `patch.FindOne()` [returns `nil, nil`](https://github.com/evergreen-ci/evergreen/blob/8c915b9282ad7b4c36a0be3182398c20f6d713d2/model/patch/db.go#L147) if it can't find a matching patch.

Reply:
No (although you should return there as well): err != nil when there's a database error finding the patch, but if you go into patch.FindOne it returns `nil, nil` if results are not found (which is .... dumb)

Reply:
Ah! thank you!